src/Systmm.f90

```diff
@@ -182,116 +181,116 @@ SUBROUTINE SYSTMM(temp_file,param_file)
-!
-!    Set NCELL0 for purposes of tributary input
-!
-          ncell0=nncell
-          dt_total=dt_calc
-          do nm=no_dt(ns),1,-1
-            z=depth(nncell)
-            call energy(T_0,q_surf,nncell)
-            q_dot=(q_surf/(z*rfac))
-            T_0=T_0+q_dot*dt_calc
-            if(T_0.lt.0.0) T_0=0.0
-!
-!     Look for a tributary.
-!
-            Q1=Q_in(nncell)
-            ntribs=no_tribs(nncell)
-            if(ntribs.gt.0.and..not.DONE) then
-              do ntrb=1,ntribs
-                nr_trib=trib(nncell,ntrb)
-                if(Q_trib(nr_trib).gt.0.0) then
-                  Q2=Q1+Q_trib(nr_trib)
-                  T_0=(Q1*T_0+Q_trib(nr_trib)*T_trib(nr_trib))/Q2
+!          dt_calc=dt_part(ns)
+          nncell=segment_cell(nr,nstrt_elm(ns))
+!
+!    Set NCELL0 for purposes of tributary input
+!
+          ncell0=nncell
+          dt_total=dt_calc
+          do nm=no_dt(ns),1,-1
+            z=depth(nncell)
+            call energy(T_0,q_surf,nncell)
+            q_dot=(q_surf/(z*rfac))
+            T_0=T_0+q_dot*dt_calc
+            if(T_0.lt.0.0) T_0=0.0
+!
+!     Look for a tributary.
+!
+            Q1=Q_in(nncell)
+            ntribs=no_tribs(nncell)
+            if(ntribs.gt.0.and..not.DONE) then
+              do ntrb=1,ntribs
+                nr_trib=trib(nncell,ntrb)
+                if(Q_trib(nr_trib).gt.0.0) then
+                  Q2=Q1+Q_trib(nr_trib)
+                  T_0=(Q1*T_0+Q_trib(nr_trib)*T_trib(nr_trib))/Q2
                 end if
-!
-                Q1=Q_out(nncell)
-!
-              end do
-              DONE=.TRUE.
-            end if
-            if(ntribs.eq.0.and.Q_diff(nncell).gt.0) then
-              Q2=Q1+Q_diff(nncell)
-              T_dist=T_head(nr)
-              T_0=(Q1*T_0+Q_diff(nncell)*T_dist)/Q2
-              Q1=Q2
-            end if
+!
+                Q1=Q_out(nncell)
+!
+              end do
+              DONE=.TRUE.
+            end if
+            if(ntribs.eq.0.and.Q_diff(nncell).gt.0) then
+              Q2=Q1+Q_diff(nncell)
+              T_dist=T_head(nr)
+              T_0=(Q1*T_0+Q_diff(nncell)*T_dist)/Q2
+              Q1=Q2
+            end if
 500 continue
-            nseg=nseg+1
-            nncell=segment_cell(nr,nseg)
-!
-!     Reset tributary flag if this is a new celln_write
-!
-            if(ncell0.ne.nncell) then
-              ncell0=nncell
-              DONE=.FALSE.
-            end if
-            dt_calc=dt(nncell)
-            dt_total=dt_total+dt_calc
+            nseg=nseg+1
+            nncell=segment_cell(nr,nseg)
+!
+!     Reset tributary flag if this is a new celln_write
+!
+            if(ncell0.ne.nncell) then
+              ncell0=nncell
+              DONE=.FALSE.
+            end if
+            dt_calc=dt(nncell)
+            dt_total=dt_total+dt_calc
+          end do
+          if (T_0.lt.0.5) T_0=0.5
+            temp(nr,ns,n2)=T_0
+	    T_trib(nr)=T_0
+!
+!   Write all temperature output UW_JRY_11/08/2013
```

Comment:
The following comment may not be accurate since it predates the upgrade of RBM to v2.0. The write statement, as written here, writes the output to every location rather than the initial cell.
JRY 1/20